shimming-toolbox/shimmingtoolbox/cli/b0shim.py

```diff
@@ -322,21 +322,33 @@ def dynamic(fname_fmap, fname_anat, fname_mask_anat, method, opt_criteria, slice
                 logger.debug("Converting Siemens scanner coil from Shim CS (LAI) to Gradient CS")
                 # First convert to RAS
                 orders = tuple([order for order in scanner_coil_order if order != 0])
-                for i_shim in range(coefs.shape[0]):
-                    # Convert coefficient
-                    coefs_coil[i_shim, 1:] = shim_to_phys_cs(coefs_coil[i_shim, 1:], manufacturer, orders)
 
-                # Convert coef of 1st order sph harmonics to Gradient coord system
-                coefs_freq, coefs_phase, coefs_slice = phys_to_gradient_cs(coefs_coil[:, 1],
-                                                                           coefs_coil[:, 2],
-                                                                           coefs_coil[:, 3], fname_anat)
+                if 0 in scanner_coil_order:
+                    for i_shim in range(coefs.shape[0]):
+                        # Convert coefficient
+                        coefs_coil[i_shim, 1:] = shim_to_phys_cs(coefs_coil[i_shim, 1:], manufacturer, orders)
+                    # Convert coef of 1st order sph harmonics to Gradient coord system
+                    coefs_freq, coefs_phase, coefs_slice = phys_to_gradient_cs(coefs_coil[:, 1],
+                                                                               coefs_coil[:, 2],
+                                                                               coefs_coil[:, 3], fname_anat)
+                    coefs_coil[:, 1] = coefs_freq
+                    coefs_coil[:, 2] = coefs_phase
+                    coefs_coil[:, 3] = coefs_slice
```

Comment:
Identical whether 0 is in orders or not. Would put it outside the if statement

Reply:
The indexes were not the same, but this was allowed to be changed by moving to a dictionary